cli/bpmetadata/path.go

```diff
@@ -0,0 +1,85 @@
+package bpmetadata
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+)
+
+const (
+	regexEx  = ".*/(examples/.*)"
+	regexMod = ".*/(modules/.*)"
+)
+
+func isPathValid(path string) (bool, error) {
+	_, err := os.ReadFile(path)
+	if err != nil {
+		return false, fmt.Errorf("unable to read file at the provided path: %v", err)
+	}
+
+	return true, nil
+}
+
+func getExamples(configPath string) ([]BlueprintMiscContent, error) {
+	re := regexp.MustCompile(regexEx)
+	return getDirPaths(configPath, re)
+}
+
+func getModules(configPath string) ([]BlueprintMiscContent, error) {
+	re := regexp.MustCompile(regexMod)
+	return getDirPaths(configPath, re)
+}
+
+func getDirPaths(configPath string, re *regexp.Regexp) ([]BlueprintMiscContent, error) {
+	dirContent, err := os.ReadDir(configPath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read dir contents: %v", err)
+	}
+
+	var paths []BlueprintMiscContent
+	for _, c := range dirContent {
+		if c.IsDir() {
```

Comment:
We can keep this left aligned by skipping if !c.IsDir() since we only care about dirs.

Reply:
ack